lightning-invoice/src/payment.rs

```diff
@@ -425,28 +496,94 @@ where
 
 		let payer = self.payer.node_id();
 		let first_hops = self.payer.first_hops();
-		let route = self.router.find_route(
-			&payer, params, &payment_hash, Some(&first_hops.iter().collect::<Vec<_>>()),
-			&self.scorer.lock()
-		).map_err(|e| PaymentError::Routing(e))?;
+		let inflight_htlcs = {
+			// Make an attempt at finding existing payment information from `payment_cache`. If it
+			// does not exist, it probably is a fresh payment and we can just return an empty
+			// HashMap.
+			if let Some(payment_info) = self.payment_cache.lock().unwrap().get(&payment_hash) {
+				let mut total_inflight_map: HashMap<(u64, bool), u64> = HashMap::new();
+
+				for hops in &payment_info.paths {
```

Comment:
May be more intuitive if you used the singular `path` instead of the plural `hops` or an element in the list.